recipes-bsp/u-boot/files/0011-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,229 @@
++	dm_gpio_request(&gpio, label) < 0 ||
++	dm_gpio_set_dir_flags(&gpio, GPIOD_IS_OUT) < 0) {
++		pr_err("IOT2050:Can not set the pin value\n");
++		return ;
++	}
++	dm_gpio_set_value(&gpio,value);
++}
++
++static int *Get_M2_Config(void)
++{
++	int i;
++	static int result[M2_CONFIG_LEN] = {0};
++	for(i=0; i < M2_CONFIG_LEN; i++)
++	{
++		result[i] = Get_PinValue(m2_key_config[i].gpio_name, m2_key_config[i].label);
++	}
++	return result;
++}
++
++static serdes_config_t Get_Serdes_Config_Set_M2_Pinmux(void)
++{
++	int i,j;
++	serdes_config_t serdes_config = {USB30, PCIE1_LANE0};//default
++	int config_table_size= sizeof(config_table)/sizeof(config_table[0]);
++
++    //1.TBD:check eeprom or flash
++    //2.configure the m2
++	int *config = Get_M2_Config();
++	printf("M2-KEY PIN:");
++	for(i=0; i < M2_CONFIG_LEN;i++)
++	{
++		printf("%d",*(config+i));
++	}
++	printf("\n");
++
++	for(i = 0; i < config_table_size; i++)
++	{
++		if(!memcmp(config, config_table[i].m2config, M2_CONFIG_LEN * sizeof(int)))
++		{
++			// set m2 pinmux
++			for(j = 0; j < PCIE_MUX_CONTROL; j++)
++				Set_PinValue(m2_pinmux_config[j].gpio_name, m2_pinmux_config[j].label, config_table[i].m2_pinmux_value[j]);
++			//get serdes config info
++			serdes_config =  config_table[i].serdes_config;
++			break;
++		}
++	}
++
++	return serdes_config;
++}
++
++
+ static bool board_is_advanced(void)
+ {
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+@@ -60,7 +195,16 @@ static bool board_is_sr1(void)
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+ 
+ 	return info->magic == IOT2050_INFO_MAGIC &&
+-		!strstr((char *)info->name, "-PG2");
++		!strstr((char *)info->name, "-PG2") &&
++		!strstr((char *)info->name, "-M2");
++}
++
++static bool board_is_m2(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strstr((char *)info->name, "-M2") != NULL;
+ }
+ 
+ static void remove_mmc1_target(void)
+@@ -76,6 +220,28 @@ static void remove_mmc1_target(void)
+ 	free(boot_targets);
+ }
+ 
++static void set_m2_info(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++	serdes_config_t serdes_conf;
++    /*IOT2050 M2 Variant*/
++	if(strcmp(info->name, "IOT2050-ADVANCED-M2") == 0) {
++		//Enable Power of E-key and B-key connector
++		Set_PinValue(m2_enable_power.gpio_name, m2_enable_power.label, 1);
++        serdes_conf = Get_Serdes_Config_Set_M2_Pinmux();
++		if (serdes_conf.lane0_sel == PCIE0_LANE0 && serdes_conf.lane1_sel == PCIE0_LANE1)
++			env_set("name_overlays", "k3-am654-iot2050-bkey-pciex2.dtbo");
++		else if (serdes_conf.lane0_sel == USB30 && serdes_conf.lane1_sel == PCIE0_LANE1)
++			env_set("name_overlays", "k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo");
++		else if (serdes_conf.lane0_sel == PCIE0_LANE0 && serdes_conf.lane1_sel == PCIE1_LANE0)
++			env_set("name_overlays", "k3-am654-iot2050-bkey-pcie-ekey-pcie.dtbo");
++		else 
++			env_set("name_overlays", "k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo");
```

Comment:
So each combination of lane0_sel and lane1_sel is 1:1 mapping to the dtbo name, seems we can use the dtbo name directly in the `static key_config_t config_table[]` so that this part could be saved.

Reply:
do not use overlay,  not exist this issue any more